volatility3/framework/plugins/linux/psscan.py

```diff
@@ -0,0 +1,158 @@
+# This file is Copyright 2023 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+#
+import logging
+from typing import Iterable, List, Tuple
+import struct
+from enum import Enum
+
+from volatility3.framework import renderers, interfaces, symbols, constants
+from volatility3.framework.configuration import requirements
+from volatility3.framework.objects import utility
+from volatility3.framework.layers import scanners
+from volatility3.framework.renderers import format_hints
+
+vollog = logging.getLogger(__name__)
+
+
+class DescExitStateEnum(Enum):
+    """Enum for linux task exit_state as defined in include/linux/sched.h"""
+
+    TASK_RUNNING = 0x00000000
+    EXIT_DEAD = 0x00000010
+    EXIT_ZOMBIE = 0x00000020
+    EXIT_TRACE = EXIT_ZOMBIE | EXIT_DEAD
+
+
+class PsScan(interfaces.plugins.PluginInterface):
+    """Scans for processes present in a particular linux image."""
+
+    _required_framework_version = (2, 0, 0)
+
+    @classmethod
+    def get_requirements(cls) -> List[interfaces.configuration.RequirementInterface]:
+        return [
+            requirements.ModuleRequirement(
+                name="kernel",
+                description="Linux kernel",
+                architectures=["Intel32", "Intel64"],
+            ),
+        ]
+
+    def _get_task_fields(
+        self, task: interfaces.objects.ObjectInterface
+    ) -> Tuple[int, int, int, str, str]:
+        """Extract the fields needed for the final output
+
+        Args:
+            task: A task object from where to get the fields.
+        Returns:
+            A tuple with the fields to show in the plugin output.
+        """
+        pid = task.tgid
+        tid = task.pid
+        ppid = task.parent.tgid if task.parent else 0
+        name = utility.array_to_string(task.comm)
+        exit_state = DescExitStateEnum(task.exit_state).name
+
+        task_fields = (
+            format_hints.Hex(task.vol.offset),
+            pid,
+            tid,
+            ppid,
+            name,
+            exit_state,
+        )
+        return task_fields
+
+    def _generator(self):
+        """Generates the tasks found from scanning."""
+
+        for task in self.scan_tasks(
+            self.context, self.config["kernel"], self.config["kernel.layer_name"]
```

Comment:
This still uses `kernel.layer_name`, sorry to be a pain, but to ensure things don't break in future it would be better to pull out the kernel and then ask it for the layer_name associated with it.